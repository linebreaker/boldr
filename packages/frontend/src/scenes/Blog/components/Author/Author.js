/* @flow */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import Link from 'react-router-dom/Link';
import { Avatar, Headline, Social } from '@boldr/ui';
import { StyleClasses } from '../../../../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.ARTICLE_SIDEBAR_AUTHOR;

const AvatarName = styled.div`
  display: inline-block;
  vertical-align: middle;
  &:first-child {
    margin-right: 15px;
  }
`;

const BioBlock = styled.div`
  vertical-align: middle;
  margin-bottom: 15px;
  padding-top: 10px;
`;

export type Props = {
  className: string,
  author: Object,
};

const Author = (props: Props) => {
  const { author } = props;
  const classes = classnames(BASE_ELEMENT, props.className);
  const authorName = (
    <Link to={`/profiles/${author.username}`}>
      {author.username}
    </Link>
  );
  return (
    <div className={classes}>
      <AvatarName>
        <Avatar src={author.avatarUrl} role="presentation" />
      </AvatarName>
      <AvatarName>
        <Headline type="h3">
          {authorName}
        </Headline>
      </AvatarName>
      <BioBlock>
        {author.bio}
      </BioBlock>
      <Social
        facebook
        fburl={author.socialMedia.facebookUrl}
        twitter
        turl={author.socialMedia.twitterUrl}
        google
        gurl={author.socialMedia.googleUrl}
        github
        ghurl={author.socialMedia.githubUrl}
        linkedin
        lurl={author.socialMedia.linkedinUrl}
      />
    </div>
  );
};
const defaultProps = {
  author: {
    username: 'Placeholder',
  },
};
Author.defaultProps = defaultProps;
export default Author;
