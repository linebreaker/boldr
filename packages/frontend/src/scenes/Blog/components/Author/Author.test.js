import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Author from './Author';

const authorProps = {
  avatarUrl: 'abcd',
  username: 'hey',
  firstName: 'me',
  lastName: 'you',
  bio: 'bio bio',
  socialMedia: {
    facebookUrl: 'http://facebook.com',
    githubUrl: 'http://facebook.com',
    twitterUrl: 'http://facebook.com',
    linkedinUrl: 'http://facebook.com',
    googleUrl: 'http://facebook.com',
    stackoverflowUrl: 'http://facebook.com',
  },
};

describe('<Author />', () => {
  const wrapper = shallow(<Author author={authorProps} />);
  it('renders <Author /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.instance().props.author.username).toBe('hey');
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
