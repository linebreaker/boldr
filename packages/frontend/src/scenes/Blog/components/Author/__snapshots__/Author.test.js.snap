// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Author /> renders snapshot 1`] = `
<div
  className="boldrui-article-sidebar__author"
>
  <styled.div>
    <Avatar
      role="presentation"
      src="abcd"
      suffixes={
        Array [
          "red",
          "pink",
          "purple",
          "deep-purple",
          "indigo",
          "blue",
          "light-blue",
          "cyan",
          "teal",
          "green",
          "light-green",
          "lime",
          "yellow",
          "amber",
          "orange",
          "deep-orange",
          "brown",
          "grey",
          "blue-grey",
        ]
      }
    />
  </styled.div>
  <styled.div>
    <Headline
      type="h3"
    >
      <Link
        replace={false}
        to="/profiles/hey"
      >
        hey
      </Link>
    </Headline>
  </styled.div>
  <styled.div>
    bio bio
  </styled.div>
  <Social
    facebook={true}
    fburl="http://facebook.com"
    ghurl="http://facebook.com"
    github={true}
    google={true}
    gurl="http://facebook.com"
    iconSize="24px"
    linkedin={true}
    lurl="http://facebook.com"
    turl="http://facebook.com"
    twitter={true}
  />
</div>
`;
